### PR TITLE
cleanup(pubsub): use `Status` factory functions

### PR DESCRIPTION
**Pull request by alevenberg**
Part of the work for https://github.com/googleapis/google-cloud-cpp/issues/14107

Found the cases using:
```
git grep -e 'StatusCode::k' --and --not -e 'StatusCode::kUnimplemented' --and --not -e 'StatusCode::kOk' -- '*.cc' ':!*_test.cc' | grep pubsub/internal
```

Then used find and replace with regex & added the header
Search
```
Status\(StatusCode::kUnknown, "(.*)"\)
```
Replace
```
internal::UnknownError("$1", GCP_ERROR_INFO())
```

Header
```
#include "google/cloud/internal/make_status.h"
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14169)

### CODE REVIEW

File: google/cloud/pubsub/internal/default_pull_lease_manager.cc [hunk 1]
```diff
@@ -98,8 +99,8 @@ future<Status> DefaultPullLeaseManager::ExtendLease(
       [stub = std::move(stub), deadline = now + extension, clock = clock_,
        impl = impl_](auto cq, auto context, auto options, auto const& request) {
         if (deadline < clock->Now()) {
-          return make_ready_future(
-              Status(StatusCode::kDeadlineExceeded, "lease already expired"));
+          return make_ready_future(internal::FailedPreconditionError(
```
Comment (coryan) [inline google/cloud/pubsub/internal/default_pull_lease_manager.cc:102]: I think `internal::DeadlineExceededError`

File: google/cloud/pubsub/internal/streaming_subscription_batch_source.cc [hunk 2]
```diff
@@ -270,7 +271,8 @@ void StreamingSubscriptionBatchSource::StartStream(
       "subscription=" + internal::UrlEncode(request.subscription()));
   auto stream = stub_->AsyncStreamingPull(cq_, std::move(context), options_);
   if (!stream) {
-    OnRetryFailure(Status(StatusCode::kUnknown, "null stream"));
+    OnRetryFailure(
+        internal::FailedPreconditionError("null stream", GCP_ERROR_INFO()));
```
Comment (coryan) [inline google/cloud/pubsub/internal/streaming_subscription_batch_source.cc:275]: `internal::UnknownError` ?

File: google/cloud/pubsub/internal/default_pull_lease_manager.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/default_pull_lease_manager.cc:102]: Agh, ok I mixed up some of the find/replace

Comment (alevenberg) [reply google/cloud/pubsub/internal/default_pull_lease_manager.cc]: Done

**Review by coryan — APPROVED** [review]

File: google/cloud/pubsub/internal/extend_leases_with_retry.cc [hunk 3]
```diff
@@ -180,7 +181,8 @@ class ExtendLeasesHandle
   CompletionQueue cq_;
   google::pubsub::v1::ModifyAckDeadlineRequest request_;
   std::unique_ptr<pubsub::BackoffPolicy> backoff_ = ExactlyOnceBackoffPolicy();
-  Status last_status_ = Status(StatusCode::kUnknown, "Exhausted before start");
+  Status last_status_ =
+      internal::UnknownError("Exhausted before start", GCP_ERROR_INFO());
```
Comment (alevenberg) [inline google/cloud/pubsub/internal/extend_leases_with_retry.cc:185]:
```suggestion
	      internal::UnknownError("Exhausted before start");
```

**Pull request merged**